modules/openthread/platform/crypto_psa.c

```diff
@@ -125,6 +130,15 @@ static void ensureKeyIsLoaded(otCryptoKeyRef aKeyRef)
 void otPlatCryptoInit(void)
 {
 	psa_crypto_init();
+
+#if !defined(CONFIG_BUILD_WITH_TFM) && defined(CONFIG_OPENTHREAD_CRYPTO_PSA)
+	/*
+	 * In OpenThread, Settings are initialized after KeyManager by default. If device uses
+	 * PSA with emulated TFM, Settings have to be initialized at the end of otPlatCryptoInit(),
+	 * to be available before storing Network Key.
+	 */
+	__ASSERT(settings_subsys_init() == 0, "Failed to initialize settings");
```

Comment:
Please double-check if you can use assert with a message on this stage of system initialization.

Reply:
Triggered assertion manually, works correctly:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/115064000/70609a72-d879-44d4-923f-0d0f8d7577f1)

Reply:
Don't use functions inside assert statements, as these are compiled out if not enabled.